### PR TITLE
Add a target for producing CI builds

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @natemcmaster

File: build/shade/_k-standard-goals.shade [hunk 1]
```diff
@@ -230,7 +239,7 @@ default SAMPLES_PROJECT_GLOB = "samples/*/project.json"
     var projectFiles = Files.Include(TEST_PROJECT_GLOB);
     foreach (var projectFile in projectFiles)
     {
-        DotnetTest(projectFile, Configuration, " --no-build");
+        DotnetTest(projectFile, Configuration, E("KOREBUILD_DOTNET_TEST_OPTIONS") + " --no-build");
```
Comment (natemcmaster) [inline build/shade/_k-standard-goals.shade:242]: Can we remove the default value from _dotnet-test.shade? 

https://github.com/aspnet/KoreBuild/blob/dev/build/shade/_dotnet-test.shade#L19

File: build/shade/_k-standard-goals.shade [hunk 2]
```diff
@@ -40,6 +40,15 @@ default SAMPLES_PROJECT_GLOB = "samples/*/project.json"
   }
 }
 
+#--for-ci
+  @{
+    Configuration = "Release";
+    E("Configuration", Configuration);
+    E("KOREBUILD_ADD_ASSEMBLY_INFO", "1");
+    E("KOREBUILD_DOTNET_PACK_OPTIONS", " --serviceable");
```
Comment (natemcmaster) [inline build/shade/_k-standard-goals.shade]: Should this be additive? What if CI has additional options set in KOREBUILD_DOTNET_PACK_OPTIONS?

File: build/shade/_k-standard-goals.shade (hunk 1 above)
Comment (pranavkm) [inline build/shade/_k-standard-goals.shade:242]: Yup. Same with pack options - those never get used and are a bit misleading.

**Comment by pranavkm** [comment]
:up: :date:

**Comment by natemcmaster** [comment]
LGTM. :ship: :it:

**Comment by natemcmaster** [comment]
Can we get this in soon? We haven't been able to run dev builds of EF for a few days now without editing our makefile to skip the repack targets.

**Pull request closed without merging**